ontotext-yasgui-web-component/src/components/run-dropdown/query-split-button.tsx

```diff
@@ -0,0 +1,170 @@
+import {TranslationService} from '../../services/translation.service';
+import {EventService} from '../../services/event-service';
+import {
+    InternalRunDropdownValueSelectedEvent
+} from '../../models/internal-events/internal-run-dropdown-value-selected-event';
+
+interface RunAction {
+    labelKey: string;
+    value: string;
+    shortcut?: string;
+}
+
+@Component({
+    tag: 'query-split-button',
+    styleUrl: 'query-split-button.scss',
+    shadow: false,
+})
+export class QuerySplitButton {
+    @Element() hostElement: HTMLElement;
+
+    @Prop() yasqe: any;
+    @Prop() translationService: TranslationService;
+    @Prop() eventService: EventService;
+
+    @State() querySplitOpen = false;
+
+    @Listen('click', {target: 'window', capture: true})
+    onWindowClick(e: MouseEvent) {
+        if (this.querySplitOpen && this.hostElement && !this.hostElement.contains(e.target as Node)) {
+            this.closeQuerySplitMenu();
+        }
+    }
+
+    @Listen('scroll', {target: 'window', capture: true, passive: true})
+    onWindowScroll() {
+        if (this.querySplitOpen) {
+            this.closeQuerySplitMenu();
+        }
+    }
+
+    @Listen('keydown', {target: 'window'})
+    onKeyDown(e: KeyboardEvent) {
+        if (e.key === 'Escape' && this.querySplitOpen) {
+            this.closeQuerySplitMenu();
+        }
+    }
+
+    @Listen('resize', {target: 'window', passive: true})
+    onWindowResize() {
+        if (this.querySplitOpen) {
+            this.closeQuerySplitMenu();
+        }
+    }
+
+    // Shortcut button blocks the dropdown when it's open, so z-index adjusted and reset accordingly
+    private updateKeyboardShortcutsButtonZIndex(zIndex: string): void {
+        const shortcutsButton = document.querySelector(
+            '.yasqe:not(.yasqe-fullscreen) .keyboard-shortcuts-dialog-button.sparql-editor-positioning') as HTMLElement;
+
+        if (shortcutsButton) {
+            shortcutsButton.style.zIndex = zIndex;
+        }
+    }
+
+    private positionDropdownMenu(): void {
+        setTimeout(() => {
+            const button = this.hostElement.querySelector('.yasqe_querySplitWrapper') as HTMLElement;
+            const dropdown = this.hostElement.querySelector('.ontotext-run-dropdown-menu.open') as HTMLElement;
+
+            if (button && dropdown) {
+                const buttonRect = this.hostElement.getBoundingClientRect();
+                const isFullscreen = this.yasqe.classList.contains('yasqe-fullscreen');
+                const buttonGroup = window.getComputedStyle(document.querySelector('.yasqe_buttons'));
+                // Convert the string value (like "10 px") to a number and default to 0 if not a number
+                const buttonGroupRight = buttonGroup.right !== 'auto' ? parseFloat(buttonGroup.right) || 0 : 0;
+
+                dropdown.style.position = 'fixed';
+                dropdown.style.right = `${(parseFloat(dropdown.style.right) || 0) + buttonGroupRight + 33}px`;
```

Comment:
Document each numeric constant used below. Don't leave it as this because it's not clear what's their meaning: 33, 16, 2